crates/wysiwyg/src/dom/dom_methods.rs

```diff
@@ -434,6 +434,73 @@ where
         self.assert_invariants();
     }
 
+    /// Returns two new subtrees as the result of splitting the Dom symetrically without mutating
+    /// itself. Also returns the new handles of node that was split.
+    ///
+    /// Only returns nodes that are modified by the split and ignores any nodes which were not
+    /// either split or contain a node that was split.
+    pub(crate) fn split_new_sub_trees(
+        &self,
+        from_handle: &DomHandle,
+        offset: usize,
+        depth: usize,
+    ) -> (DomNode<S>, DomHandle, DomNode<S>, DomHandle) {
+        let mut clone = self.clone();
+        let right = clone.split_sub_tree(from_handle, offset, depth);
+
+        // Remove unmodified children of the right split
+        // TODO: create a utility for this
+        let mut right = DomNode::Container(ContainerNode::new(
+            S::default(),
+            ContainerNodeKind::Generic,
+            None,
+            vec![right
```

Comment:
Wouldn't `let mut right = clone.split...` already give you what you need? If I'm not mistaken, right now `split_sub_tree` already returns a generic container.

Reply:
I've updated this to use the same container type that's returned from `split_sub_tree`: [`41029eb`](https://github.com/matrix-org/matrix-rich-text-editor/pull/415/commits/41029eb8bab9240631b2ab4789b9b4bbb4252fb7)

Reply:
I was actually wrong here, I thought you could re-use the result of calling `split_sub_tree` but it's not possible to do so here. The new version is more readable, thanks!